src/sql/workbench/services/notebook/browser/models/cell.ts

```diff
@@ -311,6 +313,9 @@ export class CellModel extends Disposable implements ICellModel {
 			this._source = newSource;
 			this.sendChangeToNotebook(NotebookChangeType.CellSourceUpdated);
 			this.cellSourceChanged = true;
+			if (this.isEditMode) {
```

Comment:
There's no better place to put this? I would expect there to be something called whenever the edit mode is changed that we could use instead of this...

Reply:
I have to save it only when they edit something, so this is the one place that I found to track that easily. Do you think there can be an issue?

Reply:
Why are you saving it only when they edit something? Not only does that make the name `lastEditMode` incorrect (if they didn't actually modify anything) but we should be respecting the last edit mode they used regardless of whether they did anything after switching to it.

Reply:
@yualan So even if they don't edit anything, if they switch to a mode and then close the editing mode, I should default to the last mode they're in ?

Reply:
I would say that's a reasonable thing to do. 

@yualan @dzsquared  @chlafreniere Thoughts?

Reply:
> Why are you saving it only when they edit something? Not only does that make the name `lastEditMode` incorrect (if they didn't actually modify anything) but we should be respecting the last edit mode they used regardless of whether they did anything after switching to it.

I understood it as the last edited mode that they used and not the last mode they are in.